benchmark/scripts/preflight.py

```
@@ -0,0 +1,229 @@
+def validate_matrix(path: Path, config: dict) -> None:
+    """Validate the common four-model, ten-task, concurrency-ten job shape."""
+    if config.get("n_concurrent_trials") != 10:
+        raise SystemExit(f"{path}: n_concurrent_trials must be 10")
+    agents = config.get("agents", [])
+    if len(agents) != 4:
+        raise SystemExit(f"{path}: expected 4 agents, found {len(agents)}")
+    datasets = config.get("datasets", [])
+    if len(datasets) != 1 or set(datasets[0].get("task_names", [])) != EXPECTED_TASKS:
+        raise SystemExit(
+            f"{path}: selected DeepSWE task set does not match the frozen 10 tasks"
+        )
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)